project/src/UI/Upgrades/UpgradeMenu.gd

```diff
@@ -17,16 +17,27 @@ func _ready() -> void:
 	cargo_button.connect("button_down", self, "select_upgrade", [Events.UpgradeChoices.CARGO])
 	mine_button.connect("button_down", self, "select_upgrade", [Events.UpgradeChoices.MINING])
 	weapon_button.connect("button_down", self, "select_upgrade", [Events.UpgradeChoices.WEAPON])
+	
+	# Connect the completion signal to the last animated child
+	$HBoxContainer.get_child($HBoxContainer.get_child_count() - 1).connect("on_ani_complete", self, "on_ani_complete")
 
+func on_ani_complete(entry: bool):
+	if(entry == false):
+		hide()
 
 func open() -> void:
+	ani(true)
 	get_tree().paused = true
 	show()
 	health_button.grab_focus()
 
+# Animate the upgrade buttons in or out
+func ani(_entry) -> void:
+	for button in $HBoxContainer.get_children():
```

Comment:
You should cache the `$HBoxContainer` node in an onready variable, like we do for all get_node calls: `onready var hbox_container := $HBoxContainer`